modules/openthread/platform/radio.c

```diff
@@ -508,19 +514,48 @@ static void openthread_handle_received_frame(otInstance *instance,
 	net_pkt_unref(pkt);
 }
 
-static void openthread_handle_frame_to_send(otInstance *instance,
-					    struct net_pkt *pkt)
+#if defined(CONFIG_OPENTHREAD_NAT64_TRANSLATOR)
+
+static otMessage *openthread_ip4_new_msg(otInstance *instance, otMessageSettings *settings)
+{
+	return otIp4NewMessage(instance, settings);
+}
+
+static otError openthread_nat64_send(otInstance *instance, otMessage *message)
+{
+	return otNat64Send(instance, message);
+}
+
+#else /* CONFIG_OPENTHREAD_NAT64_TRANSLATOR */
+
+static otMessage *openthread_ip4_new_msg(otInstance *instance, otMessageSettings *settings)
+{
+	return NULL;
+}
+
+static otError openthread_nat64_send(otInstance *instance, otMessage *message)
 {
+	return OT_ERROR_DROP;
+}
+
+#endif /* CONFIG_OPENTHREAD_NAT64_TRANSLATOR */
+
+static void openthread_handle_frame_to_send(otInstance *instance, struct net_pkt *pkt)
+{
+	otError error;
 	struct net_buf *buf;
 	otMessage *message;
 	otMessageSettings settings;
+	bool is_ip6 = PKT_IS_IPv6(pkt);
 
-	NET_DBG("Sending Ip6 packet to ot stack");
+	NET_DBG("Sending %s packet to ot stack", is_ip6 ? "IPv6" : "IPv4");
 
 	settings.mPriority = OT_MESSAGE_PRIORITY_NORMAL;
 	settings.mLinkSecurityEnabled = true;
-	message = otIp6NewMessage(instance, &settings);
-	if (message == NULL) {
+
+	message = is_ip6 ? otIp6NewMessage(instance, &settings)
+			 : openthread_ip4_new_msg(instance, &settings);
+	if (!message) {
```

Comment:
could you add a `NET_ERR` message here as well?

Reply:
Added new `NET_ERR` message when memory allocation fails.